src/ecdsa/ecdh.py

```diff
@@ -0,0 +1,123 @@
+"""
+Class for performing Elliptic-curve Diffie-Hellman (ECDH) operations.
+"""
+
+from .util import number_to_string
+from .ellipticcurve import INFINITY
+from .keys import SigningKey, VerifyingKey
+
+
+__all__ = ["ECDH", "NoKeyError", "InvalidCurveError",
+           "InvalidSharedSecretError"]
+
+
+class NoKeyError(AttributeError):
+    """ECDH. Key not found but it is needed for operation."""
+
+    pass
+
+
+class NoCurveError(AttributeError):
+    """ECDH. Curve not set but it is needed for operation."""
+
+    pass
+
+
+class InvalidCurveError(RuntimeError):
+    """ECDH. Raised in case the public and private keys use different curves."""
+
+    pass
+
+
+class InvalidSharedSecretError(RuntimeError):
+    """ECDH. Raised in case the shared secret we obtained is an INFINITY."""
+
+    pass
+
+
+class ECDH(object):
+    """
+    Elliptic-curve Diffie-Hellman (ECDH). A key agreement protocol.
+    Allows two parties, each having an elliptic-curve public-private key 
+    pair, to establish a shared secret over an insecure channel
+    """""
+
+    def __init__(self, curve=None, private_key=None, public_key=None):
+        self.curve = curve
+        self.private_key = None
+        self.public_key = None
+        if private_key:
+            self.load_private_key(private_key)
+        if public_key:
+            self.load_received_public_key(public_key)
+
+    def _get_shared_secret(self, remote_public_key):
+        if self.private_key is None:
+            raise NoKeyError(
+                "Private key needs to be set to create shared secret")
```

Comment:
why it needs to be set and can't be generated?

Reply:
there are 2 cases: with key generation and key import. and this 2 cases can be easily messed up. and because of that i think it needs to make it explicit.

Reply:
ok